systems/api/api-gateway/pkg/client/client.go

```diff
@@ -0,0 +1,322 @@
+package client
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/ukama/ukama/systems/common/rest"
+	"github.com/ukama/ukama/systems/common/types"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	failedRequestMsg  = "invalid content. request has failed"
+	pendingRequestMsg = "partial content. request is still ongoing"
+)
+
+type Client interface {
+	GetNetwork(string) (*NetworkInfo, error)
+	CreateNetwork(string, string, []string, []string, float64, float64, uint32, bool) (*NetworkInfo, error)
+
+	GetPackage(string) (*PackageInfo, error)
+	AddPackage(string, string, string, string, string, string, bool, bool, int64, int64, int64, string,
+		string, string, string, string, uint64, float64, float64, float64, uint32, []string) (*PackageInfo, error)
+
+	GetSim(string) (*SimInfo, error)
+	ConfigureSim(string, string, string, string, string, string, string, string, string, string,
+		string, string, string, string, uint32) (*SimInfo, error)
+
+	GetNode(string) (*NodeInfo, error)
+	RegisterNode(string, string, string, string) (*NodeInfo, error)
+	AttachNode(string, string, string) error
+	DetachNode(string) error
+	AddNodeToSite(string, string, string) error
+	RemoveNodeFromSite(string) error
+	DeleteNode(string) error
+}
+
+type clients struct {
```

Comment:
I think right way would be have a interface per client Like interfca efor network, package, sim etc.
Rahther than haveing one cleint interface. May it could be useful at somepoint of time when we try to move all these interfaces as common package.

Reply:
Addressed